spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -106,4 +129,80 @@ private String constructSparkParameters(String datasourceName) throws URISyntaxE
         "spark.sql.catalog." + datasourceName, FLINT_DELEGATE_CATALOG);
     return s3GlueSparkSubmitParameters.toString();
   }
+
+  private StartJobRequest getStartJobRequestForNonIndexQueries(
+      DispatchQueryRequest dispatchQueryRequest) {
+    StartJobRequest startJobRequest;
+    FullyQualifiedTableName fullyQualifiedTableName =
+        SQLQueryUtils.extractFullyQualifiedTableName(dispatchQueryRequest.getQuery());
+    if (fullyQualifiedTableName.getDatasourceName() == null) {
```

Comment:
Do we need to add check for query with multiple table name present? I assume we don't support it for now?

Reply:
yes, I am assuming a single table name. Need to handle it in a separate PR.
Will make a TODO comment in SQLQueryUtils.